src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1965,6 +1965,11 @@ private IConstantPatternOperation CreateBoundConstantPatternOperation(BoundConst
         private IDeclarationPatternOperation CreateBoundDeclarationPatternOperation(BoundDeclarationPattern boundDeclarationPattern)
         {
             ISymbol variable = boundDeclarationPattern.Variable;
+            if (variable == null && boundDeclarationPattern.VariableAccess.Kind == BoundKind.DiscardExpression)
+            {
+                variable = (IDiscardSymbol)(((BoundDiscardExpression)boundDeclarationPattern.VariableAccess).ExpressionSymbol);
```

Comment:
Is the cast to `(IDiscardSymbol)` necessary?

Reply:
No, I'll remove it.